gordo_components/server/server.py

```diff
@@ -123,10 +133,77 @@ def base_healthcheck():
     return app
 
 
-def run_server(host: str = "0.0.0.0", port: int = 5555, debug: bool = False):
-    app = build_app()
-    app.run(host, port, debug=debug, threaded=False)
+def run_server(
+    host: str, port: int, workers: int, worker_connections: int, log_level: str
+):
+    """
+    Run application with Gunicorn server using Gevent Async workers
+
+    Parameters
+    ----------
+    host: str
+        The host to run the server on.
+    port: int
+        The port to run the server on.
+    workers: int
+        The number of worker processes for handling requests.
+    worker_connections: int
+        The maximum number of simultaneous clients per worker process.
+    log_level: str
+        The log level for the server.
+    """
 
+    try:
+        ipaddress.ip_address(host)
+    except Exception:
+        raise
+
+    if not (1 <= port <= 65535):
+        raise ValueError(f"{port} is not a valid port number")
+
+    gunicorn_log_levels = Logger.LOG_LEVELS.keys()
+    if log_level not in gunicorn_log_levels:
+        raise ValueError(
+            f"'{log_level}' log level is not support by Gunicorn. Must be one of {', '.join(gunicorn_log_levels)}."
+        )
```

Comment:
I propose you instead leave the validation to `click`

ie.
```python
@click.option('--log-level', type=click.Choice(Logger.LOG_LEVELS.keys(), case_sensitive=False))
```

Reply:
Brilliant!